### ISSUE TITLE
Very slow performance

### ISSUE THREAD

**Issue by SamMousa** [surveyjs/survey-analytics#89]
Tested analytics package with ~23.000 records.
Survey contains ~32 questions

Interacting takes >5 seconds, sometimes memory usage spikes to 4.5GB.

Will look into narrowing this down by disabling certain question types, but currently this not at the plug and play stage yet ;-)

**Comment by tsv2013** [comment]
Can you share any details like survey JSON? Are you using summary view or table representation?

**Comment by SamMousa** [comment]
This is a customer;s survey so unfortunately I can't share it.

I'm using the summary view (that's the default one with the graphs right?)

I'm getting OOM errors from wordcloud, but can't find how to disable wordcloud visualization in the docs.

**Comment by tsv2013** [comment]
We need to know some details like question types and choices count, error messages and sample texts to reproduce issues locally and investigate the errors. I believe you understand me.

If you force the wordcloud widget to analyze 23.000 copies of encyclopedia Britannica, then 5 seconds is a great result ;-)

**Comment by SamMousa** [comment]
Fair enough, and I want to narrow it down, but I have little control due to lack of docs.
If I can disable wordcloud I can test the rest at least.

With respect to wordcloud, what metrics are you interested in?
Also the filtering should probably happen in a web worker, that way you can benefit from multiple threads and you don't need lock the UI thread. I don't necessarily mind that things take a while, but freezing the whole browser doesn't feel right :-)

**Comment by tsv2013** [comment]
I'll add the "unregisterVisualizer" method and you can get rid of wordcloud. I'll update this thread.

As for metrics - I'd like to have any starting point (probably the most painful) you can provide.

**Comment by SamMousa** [comment]
After reading this https://www.niemanlab.org/2011/10/word-clouds-considered-harmful/ I'm going to not use word clouds at all for automated reporting.

Will test without wordclouds when you have added the above feature!

**Comment by SamMousa** [comment]
I have been doing some research on how to possible use web workers to improve performance.
Below is a gist that implements a `PromiseWorker` that essentially takes work pages, executes them in a web worker and returns a promise that resolves to the result.

https://gistpreview.github.io/?bcbd331d235b3b650977a919a0bade44

Using something like this unlocks more power due to multi-threading and it allows the UI thread to never be blocked.

**Comment by tsv2013** [comment]
I've added methods unregestering visualizers:
```JS
  VisualizationManager.unregisterVisualizer("text", WordCloud);
```
or
```JS
VisualizationManager.unregisterVisualizerForAll(WordCloud);
```

They will be available in the nearest update that will be released this week
